packages/block-editor/src/components/block-list/block.js

```diff
@@ -625,6 +626,9 @@ const applyWithSelect = withSelect(
 			__unstableGetBlockWithoutInnerBlocks,
 			isNavigationMode,
 		} = select( 'core/block-editor' );
+
+		const { isEditedPostEmpty } = select( 'core/editor' );
```

Comment:
Thanks for the Pr 👍 

We shouldn't be using the `core/editor` package in a generic module like `block-editor` that can be used to power any block editor (there no post sometimes). 

Do you think something like that would work?

 - Hide the appender from the first block of any block list?

Reply:
That would work in this case, but if you have any other block as your first block, like a pre-defined template, then it fails the purpose of allowing the user to add a block before.

Reply:
There's `wp.data.select('core/block-editor').getBlockCount()` but it returns 1 if there's an empty paragraph block in the editor.

Reply:
> That would work in this case, but if you have any other block as your first block, like a pre-defined template, then it fails the purpose of allowing the user to add a block before.

we could combine the check with `isEmptyDefaultBlock`

Reply:
@youknowriad I've updated the PR. Can you check if that's alright?